helix-lsp/src/client.rs

```diff
@@ -42,11 +42,13 @@ pub struct Client {
 
 impl Client {
     #[allow(clippy::type_complexity)]
+    #[allow(clippy::too_many_arguments)]
```

Comment:
I've searched the codebase and noticed we already use this flag in a few places so I took the liberty to add it here as well.